go/contacts/phonebook.go

```diff
@@ -70,13 +70,33 @@ type assertionToNameCache struct {
 
 const assertionToNameCurrentVer = 1
 
-func ResolveAndSaveContacts(mctx libkb.MetaContext, provider ContactsProvider, contacts []keybase1.Contact) (err error) {
+func ResolveAndSaveContacts(mctx libkb.MetaContext, provider ContactsProvider, contacts []keybase1.Contact) (newlyResolved []keybase1.ProcessedContact, err error) {
 	results, err := ResolveContacts(mctx, provider, contacts, keybase1.RegionCode(""))
 	if err != nil {
-		return err
+		return nil, err
 	}
+
+	// find newly resolved
 	s := mctx.G().SyncedContactList
-	return s.SaveProcessedContacts(mctx, results)
+	cts, err := s.RetrieveContacts(mctx)
+	if err == nil {
+		unres := make(map[string]struct{})
+		for _, contact := range cts {
+			if !contact.Resolved {
+				unres[contact.Assertion] = struct{}{}
+			}
+		}
+
+		for _, result := range results {
+			if _, ok := unres[result.Assertion]; ok && result.Resolved {
+				newlyResolved = append(newlyResolved, result)
+			}
+		}
```

Comment:
It might be helpful to have more descriptive names of variables in this function so it's easier to unpack.
`results` -> `resolveResults`
`cts` -> `currentContacts`